### PR TITLE
Quarkus LTS Upgrade to 3.8.4

### PR DESCRIPTION
**Pull request by porcelli**
Related https://github.com/apache/incubator-kie-issues/issues/958

Ensemble:
https://github.com/apache/incubator-kie-drools/pull/5853
https://github.com/apache/incubator-kie-optaplanner/pull/3076/files
https://github.com/apache/incubator-kie-kogito-runtimes/pull/3477
https://github.com/apache/incubator-kie-kogito-apps/pull/2039
https://github.com/apache/incubator-kie-kogito-examples/pull/1908
https://github.com/apache/incubator-kie-kogito-images/pull/1760

### CODE REVIEW

**Review by mariofusco — APPROVED** [review]

**Review by yesamer — APPROVED** [review]

File: drools-quarkus-extension/drools-quarkus-deployment/pom.xml [hunk 1]
```diff
@@ -178,29 +178,6 @@
...
-            <version>3.1.1</version>
-            <executions>
-              <execution>
-                <id>Run Script</id>
-                <phase>prepare-package</phase>
-                <goals>
-                  <goal>java</goal>
-                </goals>
-                <configuration>
-                  <mainClass>org.drools.util.RemoveCommentsMain</mainClass>
```
Comment (baldimir) [inline drools-quarkus-extension/drools-quarkus-deployment/pom.xml:196]: If this workaround is removed, the class RemoveCommentsMain could be removed from the drools-util too, as I expect the workaround will be removed also from other repositories, where used.

Comment (porcelli) [reply drools-quarkus-extension/drools-quarkus-deployment/pom.xml:196]: good point, I'll remove and update the PR

Comment (porcelli) [reply drools-quarkus-extension/drools-quarkus-deployment/pom.xml:196]: just pushed the change for that

File: drools-util/src/main/java/org/drools/util/RemoveCommentsMain.java [hunk 2]
```diff
@@ -1,51 +0,0 @@
...
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.NoSuchFileException;
-import java.nio.file.Path;
-import java.util.stream.Collectors;
-
-public class RemoveCommentsMain {
```
Comment (porcelli) [inline drools-util/src/main/java/org/drools/util/RemoveCommentsMain.java:12]: @baldimir fyi

File: drools-util/src/test/java/org/drools/util/RemoveCommentsTest.java [hunk 3]
```diff
@@ -1,15 +0,0 @@
-package org.drools.util;
-
```
Comment (porcelli) [inline drools-util/src/test/java/org/drools/util/RemoveCommentsTest.java:2]: @baldimir fyi

**Review by baldimir — APPROVED** [review]

**Comment by jstastny-cz** [comment]
Change in build-chain config won't take effect in the jenkins PR check, so it will keep failing, GHA PR checks to be considered here.

**Comment by jstastny-cz** [comment]
A note here - due to length of the build the jenkins PR runs downstream projects in "quickly" mode, so it's rather a compile downstream build - that's why there's fail in kogito-apps GHA, but jenkins PR check is green. The GHA failure is relevant and should be considered.

**Review by pefernan — APPROVED** [review]

**Pull request merged**